tests/test_sso.py

```diff
@@ -417,190 +268,231 @@ def test_get_profile_and_token_returns_expected_workosprofile_object(
+        assert profile_and_token.profile.dict() == mock_magic_link_profile
+
+    async def test_get_profile(self, mock_profile, mock_http_client_with_response):
+        mock_http_client_with_response(self.http_client, mock_profile, 200)
+
+        profile = await self.sso.get_profile("123")
+
+        assert profile.dict() == mock_profile
+
+    async def test_get_connection(
+        self, mock_connection, mock_http_client_with_response
     ):
-        connections = mock_connections_with_default_limit_v2
+        mock_http_client_with_response(self.http_client, mock_connection, 200)
 
-        mock_request_method("get", mock_connections_pagination_response, 200)
-        all_connections = connections.auto_paging_iter()
+        connection = await self.sso.get_connection(connection_id="connection_id")
 
-        number_of_connections = len(*list(all_connections))
-        assert number_of_connections == len(mock_connections["data"])
+        assert connection.dict() == mock_connection
 
-    def test_list_connections_honors_limit(
-        self,
-        mock_connections_with_limit,
-        mock_connections_pagination_response,
-        mock_request_method,
-        setup_with_client_id,
+    async def test_list_connections(
+        self, mock_connections, mock_http_client_with_response
     ):
-        connections = mock_connections_with_limit
-        mock_request_method("get", mock_connections_pagination_response, 200)
-        all_connections = SSO.construct_from_response(connections).auto_paging_iter()
+        mock_http_client_with_response(self.http_client, mock_connections, 200)
 
-        assert len(*list(all_connections)) == len(mock_connections_with_limit["data"])
+        connections = await self.sso.list_connections()
 
-    def test_list_connections_honors_limit_v2(
-        self,
-        mock_connections_with_limit_v2,
-        mock_connections_pagination_response,
-        mock_request_method,
-        setup_with_client_id,
+        assert list_data_to_dicts(connections.data) == mock_connections["data"]
+
+    async def test_list_connections_with_connection_type(
+        self, mock_connections, capture_and_mock_http_client_request
     ):
-        connections = mock_connections_with_limit_v2
-        mock_request_method("get", mock_connections_pagination_response, 200)
-        all_connections = connections.auto_paging_iter()
-        dict_mock_connections_with_limit = mock_connections_with_limit_v2.to_dict()
+        request_args, request_kwargs = capture_and_mock_http_client_request(
+            http_client=self.http_client,
+            response_dict=mock_connections,
+            status_code=200,
+        )
 
-        assert len(*list(all_connections)) == len(
-            dict_mock_connections_with_limit["data"]
+        await self.sso.list_connections(connection_type="GenericSAML")
+
+        request_params = request_kwargs["params"]
+        assert request_params["connection_type"] == "GenericSAML"
```

Comment:
For tests like this I wonder if we should explicitly assert all the params, like `assert {"connection_type": "GenericSaml", "limit": 10} == request_kwargs["params"]` or something similar. We know exactly how we're calling `list_connections`. it seems like we should know exactly how that translates into the HTTP request.